### PR TITLE
[codegen/python] add additional codegen test to cover type reference expectations

### LINKED ISSUE #12009: [python] - MLC results in incorrect usage of google-native python package
### What happened?

When building a MLC using https://www.pulumi.com/registry/packages/google-native/api-docs/container/v1/nodepool/#nodepoolautoscaling as an input to a user-defined type, the resulting python sdk incorrectly references the generated type, resulting in a runtime error.

### Expected Behavior

The generated python sdk should reference the above referenced type by:
```
 @property
    @pulumi.getter(name="locationPolicy")
    def location_policy(self) -> Optional['pulumi_google_native.container.v1.NodePoolAutoscalingLocationPolicy']:
        """
        Location policy used when scaling up a nodepool.
        """
        return pulumi.get(self, "location_policy")
```

### Steps to reproduce

Instead, the namespace appears to be truncated, resulting in incorrect usage of the imported package. **see _container below**
```
    @property
    @pulumi.getter(name="locationPolicy")
    def location_policy(self) -> Optional['_container.v1.NodePoolAutoscalingLocationPolicy']:
        """
        Location policy used when scaling up a nodepool.
        """
        return pulumi.get(self, "location_policy")
```

Repro can be referenced:
- generated python sdk :: https://github.com/phillipedwards/pulumi-aws-s3-replicated-bucket/blob/pedwards/gcp-issue/sdk/python/pulumi_aws_s3_replicated_bucket/gcp/gke/outputs.py
- schema.json :: https://github.com/phillipedwards/pulumi-aws-s3-replicated-bucket/blob/3224fb11b53d6b58825d3da1c31b22164f37b945/schema.json#L37-L46

### Output of `pulumi about`

```bash
▶ pulumi about         
CLI          
Version      3.53.0
Go Version   go1.19.5
Go Compiler  gc

Host     
OS       ubuntu
Version  22.04
Arch     x86_64

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/phillipedwards
User           phillipedwards
```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by dixler**
# Description

Fixes #12009

#12009 was fixed by https://github.com/pulumi/pulumi/pull/13584. This PR adds a test to cover the expected behavior.

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/codegen/testing/test/testdata/enum-reference/python/pulumi_example/gcp/gke/outputs.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            pulumi.set(__self__, "location_policy", location_policy)
+
+    @property
+    @pulumi.getter(name="locationPolicy")
+    def location_policy(self) -> Optional['pulumi_google_native.container.v1.NodePoolAutoscalingLocationPolicy']:
+        """
+        Location policy used when scaling up a nodepool.
+        """
+        return pulumi.get(self, "location_policy")
+
```
Comment (dixler) [inline pkg/codegen/testing/test/testdata/enum-reference/python/pulumi_example/gcp/gke/outputs.py:51]: From https://github.com/pulumi/pulumi/issues/12009#issue-1560529579

### Expected Behavior

The generated python sdk should reference the above referenced type by:
```
 @property
    @pulumi.getter(name="locationPolicy")
    def location_policy(self) -> Optional['pulumi_google_native.container.v1.NodePoolAutoscalingLocationPolicy']:
        """
        Location policy used when scaling up a nodepool.
        """
        return pulumi.get(self, "location_policy")
```

**Review by Frassle — APPROVED** [review]

**Comment by dixler** [comment]
bors merge

**Comment by Zaid-Ajaj** [comment]
bors merge

**Comment by dixler** [comment]
bors merge

**Comment by dixler** [comment]
bors cancel

**Comment by dixler** [comment]
bors merge

**Comment by dixler** [comment]
bors merge

**Pull request merged**